pkg/vpcmodel/responsive_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package vpcmodel
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/netp"
+)
+
+func newTCPConn(t *testing.T, srcMinP, srcMaxP, dstMinP, dstMaxP int64) *connection.Set {
+	t.Helper()
+	return connection.TCPorUDPConnection(netp.ProtocolStringTCP, srcMinP, srcMaxP, dstMinP, dstMaxP)
+}
+
+func newUDPConn(t *testing.T, srcMinP, srcMaxP, dstMinP, dstMaxP int64) *connection.Set {
+	t.Helper()
+	return connection.TCPorUDPConnection(netp.ProtocolStringUDP, srcMinP, srcMaxP, dstMinP, dstMaxP)
+}
+
+func newICMPconn(t *testing.T) *connection.Set {
+	t.Helper()
+	return connection.ICMPConnection(
+		connection.MinICMPType, connection.MaxICMPType,
+		connection.MinICMPCode, connection.MaxICMPCode)
+}
+
+func newTCPUDPSet(t *testing.T, p netp.ProtocolString) *connection.Set {
+	t.Helper()
+	return connection.TCPorUDPConnection(p,
+		connection.MinPort, connection.MaxPort,
+		connection.MinPort, connection.MaxPort)
+}
+
+type responsiveTest struct {
+	name     string
+	srcToDst *connection.Set
+	dstToSrc *connection.Set
+	// expectedResponsiveConn represents the subset from srcToDst which is not related to the
+	// "non-responsive" mark (*) on the srcToDst connection,
+	expectedResponsiveConn *connection.Set
```

Comment:
consider convert to `detailedConn` instead of plain connection object

Reply:
Yep, makes sense. This will  strengthen the tests

Reply:
done